docs/design/storage_classes.md

```diff
@@ -0,0 +1,413 @@
+    understanding its technical properties
+
+## Constraints and Assumptions
+
+The proposed design should:
+
+1.  Deal with manually-created volumes
+2.  Not require users to understand the differences between storage classes
+3.  Allow administrators to describe their own taxonomy of classes
+4.  Eventually allow advanced dynamic provisioning features to be created
+    based on storage classes, eg: configure the provisioner for the class by
+    adding provisioner parameters to the class itself
+
+We will focus **only** on the barest mechanisms to describe and implement
+storage classes in this proposal.  We will address the following topics in
+future proposals:
+
+1.  Dynamically provisioning new volumes for a storage class
+2.  Allowing administrators to control access to storage classes
+
+## Use Cases
+
+1.  As an administrator, I want to create a taxonomy of storage classes which
+    models the different combinations of (volume type, performance
+    characteristics) of the persistent volumes available in my cluster
+2.  As a user, I want to be able to make a claim on a persistent volume by
+    specifying the storage class as well as the currently available attributes
+
+### Use Case: Taxonomy of Storage Classes
+
+Kubernetes offers volume types for a variety of storage systems.  Within each
+of those storage systems, there are numerous ways in which volume instances
+may differ from one another: iops performance, retention policy, etc.
+Administrators of real clusters typically need to manage a variety of
+different volumes whose characteristics correspond to some logical class of
+their own devising.
+
+Kubernetes should make it possible for administrators to flexibly model the
+taxonomy of volumes in their clusters and to label volumes with their storage
+class.  This capability must be optional and fully backward-compatible with
+the existing API.
+
+As an example, there are four different types of AWS EBS volume (in ascending
+order of performance):
+
+1.  Cold HDD
+2.  Throughput optimized HDD
+3.  General purpose SSD
+4.  Provisioned IOPS SSD
+
+Currently, there is no way to distinguish between a group of 4 PVs where each
+volume is of one of these different types.  Administrators need the ability to
+distinguish between these instances and to give their own names to the
+different types.  An administrator might decide to name the classes of the
+volumes as follows:
+
+1.  Cold HDD - `tin`
+2.  Throughput optimized HDD - `bronze`
+3.  General purpose SSD - `silver`
+4.  Provisioned IOPS SSD - `gold`
+
+This is not the only dimension that EBS volumes can differ in.  Let's simplify
+things and imagine that AWS has two availability zones, `east` and `west`. Our
+administrators want to differentiate between volumes of the same type in these
+two zones, so they create a taxonomy of volumes like so:
+
+1.  `tin-west`
+2.  `tin-east`
+3.  `bronze-west`
+4.  `bronze-east`
+5.  `silver-west`
+6.  `silver-east`
```

Comment:
I agree -- I only presented it as an example of different individuals having different thought processes.  I can make a prettier example if you like; that will not be a huge problem.